extensions/amp-animation/0.1/web-animation-types.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+/**
+ * @typedef {!WebMultiAnimationDef|!WebKeyframeAnimationDef}
+ */
+export let WebAnimationDef;
+
+
+/**
+ * @mixes WebAnimationTimingDef
+ * @typedef {{
+ *   animations: !Array<!WebAnimationDef>,
+ * }}
+ */
+export let WebMultiAnimationDef;
+
+
+/**
+ * @mixes WebAnimationTimingDef
+ * @typedef {{
+ *   target: (string|!Element),
+ *   keyframes: !WebKeyframesDef,
+ * }}
+ */
+export let WebKeyframeAnimationDef;
+
+
+/**
+ * @typedef {!Object<string, *>|!Array<!Object<string, *>>}
```

Comment:
Hmm. Good question. In other words, you'd like to enumerate all allowed right away? I was also going to include `visibility` and `background` to start with.